contrib/dockerfile_build.bzl

```diff
@@ -0,0 +1,122 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Repository rule for building a Docker image from a Dockerfile and saving it
+   into a tar file.
+
+This rule uses the docker tool installed in the system to build the image and
+then save it as a tar file. The saved tar file is available for other rules to
+use (e.g. container_image, container_load, container_test)
+
+The built docker image is always called after the dockerfile_image target used
+to build the image and tagged with `dockerfile_image` (i.e. <target_name>:dockerfile_image).
+The produced tar file is available at @<repo_name>//image:dockerfile_image.tar
+"""
+
+def _docker(repository_ctx):
+    """Resolves the docker path.
+
+    Args:
+      repository_ctx: The repository context
+
+    Returns:
+      Tha path to the docker tool
+    """
+
+    if repository_ctx.attr.docker_path:
+        return repository_ctx.attr.docker_path
+    if repository_ctx.which("docker"):
+        return repository_ctx.which("docker")
+
+    fail("Path to the docker tool was not provided and it could not be resolved automatically.")
+
+def _impl(repository_ctx):
+    """Core implementation of dockerfile_image."""
+
+    docker_path = _docker(repository_ctx)
+
+    # TODO(alex1545): once the read() function is available, enable copying
```

Comment:
use cat then to read the contents, then use repository_ctx.file() to write them. Example cat use: https://github.com/bazelbuild/rules_docker/blob/master/container/pull.bzl#L189

Reply:
Done